### ISSUE TITLE
Update websockets to 10.0

### ISSUE THREAD

**Issue by VladLupashevskyi** [ethereum/web3.py#2178]
Currently it's not possible to use web3.py webscokets providers with python 3.10, because when trying to connect it fails with:

```
TypeError: As of 3.10, the *loop* parameter was removed from Lock() since it is no longer necessary
```

Solution would be: websockets package needs to be upgraded to 10.0, which support python 3.10

Related #2172

**Comment by fselmo** [comment]
Thanks for this! There is a PR open for python 3.10 support, we can start tracking changes there #2175

**Comment by DragonDmoney** [comment]
I have the same error as well with the latest version of web3.py.

**Comment by fselmo** [comment]
closed by #2324 !
